Libraries/src/Amazon.Lambda.KinesisEvents/README.md

```diff
@@ -36,3 +36,86 @@ public class Function
     }
 }
 ```
+
+The following is a sample class and Lambda function that receives Amazon Kinesis event record data as an input and uses `StreamsEventResponse` object to return batch item failures, if any. (Note that by default anything written to Console will be logged as CloudWatch Logs events.)
+
+```csharp
+public class Function
+{
+    public StreamsEventResponse Handler(KinesisEvent kinesisEvent)
+    {
+        var batchItemFailures = new List<StreamsEventResponse.BatchItemFailure>();
+        string curRecordSequenceNumber = string.Empty;
+
+        foreach (var record in kinesisEvent.Records)
+        {
+            try
+            {
+                //Process your record
+                var kinesisRecord = record.Kinesis;
+                curRecordSequenceNumber = kinesisRecord.SequenceNumber;
+            }
+            catch (Exception e)
+            {
+                /* Since we are working with streams, we can return the failed item immediately.
+                   Lambda will immediately begin to retry processing from this failed item onwards. */
+                batchItemFailures.Add(new StreamsEventResponse.BatchItemFailure() { ItemIdentifier = curRecordSequenceNumber });
+                return new StreamsEventResponse() { BatchItemFailures = batchItemFailures };
+            }
+        }
+
+        return new StreamsEventResponse() { BatchItemFailures = batchItemFailures };
+    }
+}
+```
+
+The following is a sample class and Lambda function that receives Amazon Kinesis event when using time windows as an input and uses `KinesisTimeWindowResponse` object to demonstrates how to aggregate and then process the final state. (Note that by default anything written to Console will be logged as CloudWatch Logs events.)

```

Comment:
nit: "demonstrates" should be "demonstrate"

Reply:
Fixed